File: demo1.py
## EXAMPLE 1 ##

# imports
from s3utils import * 
from AWSBatchUtils import *


# Step 1: Define parallel tasks
def BosonTask(x):
	return( x * x )

BosonParams = []
for x in range(100):
	BosonParams.append( {'x': x} )


# Step 2: Upload inputs and submit a PyBoson batch
batch_id = 1
njobs    = 10
s3_path  = 's3://boson-base/pyboson-test/'

SaveObjectesInS3(s3_path,
	key = 'batch_{}_in'.format(batch_id),
	FUN = BosonTask, X = BosonParams
)

SubmitBatchJobs(batch_id = batch_id, njobs = njobs, s3_path = s3_path)


# Step 3: Collect results
out_all = LoadObjectsFromS3(
	s3_path = s3_path,
	keys = ['batch_{}_out_'.format(batch_id) + s for s in map(str,range(1,njobs+1))]
)
print(out_all)

File: demo1.R
## EXAMPLE 1 ##

# imports
source('s3utils.R')
source('AWSBatchUtils.R')


# Step 1: Define tasks
BosonTask = function (x) {
	return ( x * x )
}

BosonParams = as.list(1:100)


# Step 2: Upload inputs and submit a RBoson batch
batch.id = 0
njobs    = 10
s3.path  = 's3://boson-base/rboson-test/'

SaveObjectesInS3(
  FUN = BosonTask, X = BosonParams,
  s3.path = s3.path,
  key = paste0('batch_', batch.id, '_in')
)

SubmitBatchJobs(batch.id = batch.id, njobs = njobs, s3.path = s3.path)


# Step 3: Collect results
out.all = LoadObjectsFromS3(
  s3.path = s3.path,
  keys = paste0('batch_', batch.id, '_out_', 1:njobs)
)